microprofile/WFLY-10709-MicroProfile-Tracing-Extension.adoc

```diff
@@ -0,0 +1,80 @@
+= [WFLY-10709] Support Eclipse MicroProfile OpenTracing
+:author:            Juraci Paixão Kröhling
+:email:             jpkroehling@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,tracing,openshift
+
+== Overview
+
+In order to improve Observability of applications deployed in WildFly it would be beneficial to provide
+transparent tracing of Java EE components via automatic initialization of a tracer (Jaeger) plus a set
+of instrumentation libraries for components commonly used in Java EE applications, such as Servlets, JAX-RS,
+CDI, EJB, ...
+
+This will be implemented by supporting Eclipse MicroProfile OpenTracing 1.1 for applications deployed in WildFly.
+The integration uses the smallrye-opentracing component to provide the MicroProfile OpenTracing implementation.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10709[WFLY-10709]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1027[EAP7-1027]
+* https://issues.jboss.org/browse/CLOUD-2728[CLOUD-2728] - Expose EAP "Tracing" information so that it can be consumed by OpenShift Console
+* https://issues.jboss.org/browse/TRACING-85[TRACING-85] - Create jboss-modules for OpenTracing and Jaeger
+* https://issues.jboss.org/browse/TRACING-87[TRACING-87] - OpenTracing integration for JAX-RS in Wildfly
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:tremes@redhat.com[Tomas Remes]
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile OpenTracing is provided by https://github.com/smallrye/smallrye-opentracing[smallrye-opentracing].
+
+=== Other Interested Projects
+
+* A https://github.com/jpkrohling/java-wildfly-tracing[prototype of a WildFly extension] for OpenTracing was developed while exploring a solution for TRACING-85/.../TRACING-88.
+
+== Requirements
+
+* Support Eclipse MicroProfile OpenTracing 1.1 for applications deployed in WildFly.
```

Comment:
@jpkrohling Are we aiming towards the supporting the tracing of JAX-RS services running inside of WildFly now? The scope is kinda unclear for now.

Reply:
I'll update the doc to make it clearer, but tracing of JAX-RS endpoints is indeed part of MicroProfile OpenTracing 1.1.